presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -268,7 +268,7 @@ private ExpressionAnalyzer(
         this.parameters = requireNonNull(parameters, "parameterMap is null");
         this.isDescribe = isDescribe;
         this.warningCollector = requireNonNull(warningCollector, "warningCollector is null");
-        this.outerScopeSymbolTypes = ImmutableMap.copyOf(requireNonNull(outerScopeSymbolTypes, "outerScopeSymbolTypes is null"));
+        this.outerScopeSymbolTypes = requireNonNull(outerScopeSymbolTypes, "outerScopeSymbolTypes is null");
```

Comment:
Nice ! How did you figure out that this copy is causing the timeout ?

Reply:
Added test. Just run locally with same query and pause debugger at random points to see what its doing